gcc/rust/ast/rust-ast.h

```diff
@@ -1831,13 +1835,34 @@ class ASTFragment
     return *this;
   }
 
-  static ASTFragment create_empty () { return ASTFragment ({}); }
   static ASTFragment create_error () { return ASTFragment ({}, true); }
 
+  static ASTFragment create_one (SingleASTNode &&node)
+  {
+    auto fragment = ASTFragment ({node}, false);
+    fragment.fragment_is_one_node = true;
+
+    return fragment;
+  }
+
   std::vector<SingleASTNode> &get_nodes () { return nodes; }
   bool is_error () const { return fragment_is_error; }
+  bool is_one_node () const { return fragment_is_one_node; }
 
-  bool should_expand () const { return !is_error () && !nodes.empty (); }
+  bool should_expand () const
+  {
+    // If the fragment is an error fragment, it should not expand
+    if (is_error ())
+      return false;
+
+    // If the fragment contains zero nodes, it might be a fragment we want to
+    // expand! In the case of statements, items, trait impl items and so on, it
+    // is perfectly valid to transcribe zero of those things.
+    if (is_one_node ())
+      return nodes.size () == 1;
```

Comment:
this seems confusing to me. If we can remove the flag then I think this is fine. But I might be missing something going on. Is this a flag to say its an empty macro invocation but we need to still transcribe? if ```!nodes.empty()```